### PR TITLE
Fixed issues with multi-model-statistics timeranges

### LINKED ISSUE #1485: `timerange` in `multi_model_statistics` output can be wrong and `start_year` and `end_year` can be missing
The calculation of the `timerange` for `multi_model_statistics` output does not consider the `span` attribute:

https://github.com/ESMValGroup/ESMValCore/blob/84d1b49ccbf83f8dbe0c2179467be583260398b4/esmvalcore/_recipe.py#L667-L682

This leads to inconsistencies between the filename and the data when `span='overlap'` is used. E.g., using these datasets

```yml
- {dataset: BCC-ESM1, project: CMIP6, exp: historical, ensemble: r1i1p1f1, grid: gn, start_year: 1920, end_year: 1930}
- {dataset: CanESM2, project: CMIP5, exp: historical, ensemble: r1i1p1, start_year: 1925, end_year: 1935}
```

with `span='overlap'` leads to a file that correctly contains the years 1925 to 1930, but is named `MultiModelMean_historical_Amon_tas_1920-1935.nc` and the `metadata.yml` is incorrect.

In addition, `start_year` and `end_year` are missing in the `metadata.yml`, which causes some diagnostics to fail.

Thanks @LisaBock for reporting this. I will add a fix soon.

### PR DESCRIPTION
**Pull request by schlunma**
## Description

This PR

- fixes an issue with integer auxiliary time coordinates in the `multi_model_statistics` preprocessor that occurs when preprocessor like `annual_statitics` are used in combination with `multi_model_statistics`.
- Adds the attributes `start_year` and `end_year` to the multi-models statistics output that is needed by many diagnostics.
- Fixes the timerange of multi-models statistics output  when `span=overlap` is used.

Closes #1485

Link to documentation: https://esmvaltool--1486.org.readthedocs.build/projects/ESMValCore/en/1486/recipe/overview.html

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by schlunma** [comment]
@LisaBock can you try if this PR fixes your problem? Thanks!!

**Comment by LisaBock** [comment]
> @LisaBock can you try if this PR fixes your problem? Thanks!!

Thanks @schlunma ! I reran my recipe and it works now. There is a start_year in the metadata for the MultiModelMean. Perfect!

File: doc/recipe/overview.rst [hunk 1]
```diff
@@ -90,11 +90,18 @@ data specifications:
...
-  year. The starting point and end point must be separated with '/'.
-  (e.g key-value ``timerange: '1982/1990'``)
+  Alternatively, the time range can be specified in `ISO 8601 format
+  <https://en.wikipedia.org/wiki/ISO_8601>`_, for both dates and periods.
+  The starting point and end point must be separated with ``/`` (e.g.,
+  ``timerange: '1982/1990'`` will select the data from the years 1982 to 1990,
+  ``timerange: '1910/P2Y'`` will select 2 years of data starting in 1910, or
+  ``P1Y2M/19201231T235959`` will select 1 year and 2 months of data ending on
+  31 December 1920 23:59:59).
+  In addition, wildcards (``'*'``) are accepted, which allow the selection of
```
Comment (sloosvel) [inline doc/recipe/overview.rst:100]: The combination of timerange values, wildcards and more examples is explained starting in line 167

Comment (schlunma) [reply doc/recipe/overview.rst:100]: Ahh, right, didn't see that! I will revert this change and add a pointer to this section

**Comment by schlunma** [comment]
Any further comments on this? I would like to merge this soon so we can test the recipes with this change :smile: Thanks!!

**Review by valeriupredoi — APPROVED** [review]
LGTM but I'll let @sloosvel have a more in-depth review, cheers Manu!

**Review by sloosvel — APPROVED** [review]
The changes in _recipe and the documentation look fine.
I ran a test with both `overlap` and `span` values and the generated preprocessor files have
the proper metadata in terms of `timerange`, `start_year` and `end_year`.
The changes in _multimodel seem to work fine, but I am not so familiar with all the details of how _multimodel works. But if it's fixing reported issues I guess it's good to go.

**Comment by schlunma** [comment]
Great, thanks for the review!

**Pull request merged**